modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/OffsetDateTime.kt.mustache

```diff
@@ -0,0 +1,89 @@
+package {{packageName}}.infrastructure
+
+import kotlinx.datetime.*
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+/**
+ * Represents a [LocalDateTime] and the respective [TimeZone] of it.
+ */
+@Serializable(with = OffsetDateTime.Companion::class)
+public data class OffsetDateTime(val dateTime: LocalDateTime, val timeZone: TimeZone) {
+    override fun toString(): String {
+        return if (timeZone == TimeZone.UTC) {
+            "${dateTime}Z"
+        } else {
+            "$dateTime$timeZone"
+        }
+    }
+
+    /**
+     * Converts the [OffsetDateTime] to an [Instant]. This looses the [TimeZone] information, because the date and time
+     * is converted to UTC in the process.
+     */
+    public fun toInstant(): Instant = dateTime.toInstant(timeZone)
+
+    /**
+     * Creates an [OffsetDateTime] from an [Instant] in a given [TimeZone] ([TimeZone.UTC] by default).
+     */
+    public fun fromInstant(instant: Instant, timeZone: TimeZone = TimeZone.UTC): OffsetDateTime = OffsetDateTime(
+        instant.toLocalDateTime(timeZone),
+        timeZone,
+    )
+
+    /**
+     * Returns a new [OffsetDateTime] with the given [TimeZone].
+     */
+    public fun inTimeZone(newTimeZone: TimeZone): OffsetDateTime {
+        val instant = dateTime.toInstant(timeZone)
+        val newDateTime = instant.toLocalDateTime(newTimeZone)
+        return OffsetDateTime(newDateTime, newTimeZone)
+    }
+
+    /**
+     * Returns a new [OffsetDateTime] with the time zone [TimeZone.UTC].
+     */
+    public fun utc(): OffsetDateTime = inTimeZone(TimeZone.UTC)
```

Comment:
Is there a use-case for this not covered by `toInstant(): Instant`? `Instant` is already intended to represent UTC timestamps.

Reply:
Well this can be used to get a local utc time for example, you can get the same info from the instant too, but this would be a shortcut.